src/mono/mono/metadata/image.c

```diff
@@ -1881,6 +1881,14 @@ mono_image_open_a_lot (MonoAssemblyLoadContext *alc, const char *fname, MonoImag
 	return mono_image_open_a_lot_parameterized (li, alc, fname, status);
 }
 
+MonoImage *
+mono_image_open_mibc (MonoAssemblyLoadContext *alc, const char *fname, MonoImageOpenStatus *status)
+{
+	MonoImage *mibcImage = mono_image_open_a_lot (alc, fname, status);
+	mibcImage->not_executable = TRUE;
```

Comment:
I'd like us to do more plumbing work here: the MonoImage is created in `do_mono_image_open`.  It would be nice if we could set `mibc->not_executable` in that function.

two reasons:
1. the loader is concurrent, so as a general practice it's better if we don't change bits in the MonoImage after it has been created and added to the cache - potentially another thread could be accessing it already.  (that seems unlikely with mibc images, but I don't want to set a precedent)
2. I hate the ton of boolean flags that we pass around everywhere - this is a good opportunity to clean things up slighlty.


That means passing an extra boolean flag down from here all the way to that function.

What I would suggest is adding a new options type (private to image.c for now).

```c
typedef struct {
   int     care_about_cli : 1;
   int     care_about_pecoff : 1;
} ImageLoadOptions;
  
typedef struct {
   ImageLoadOptoins load_options;
   int     metadata_only : 1;
   int     not_executable : 1;
} ImageOpenOptions;

static MonoImage *
do_mono_image_open (MonoAssemblyLoadContext *alc, const char *fname, MonoImageOpenStatus *status,  const MonoImageOpenOptions *options)
{
   ...
   MonoImage *image = g_new0 (MonoImage, 1);
   image->metadata_only = !!options->metadata_only;
   image->not_executable = !!options->not_executable;
   ...
   /* pass the load options to do_mono_image_load */
   return do_mono_image_load (image, status, &options->load_optoins);
}

static MonoImage *
do_mono_image_load (MonoImage *image, MonoImageOpenStatus *status,
                                     const ImageLoadOptions *options)
{
   ...
   /* same as before, but use options->care_about_cli and options->care_about_pecoff */
}
```

And then the callers would do things like:
```
MonoImage *
mono_pe_file_open (const char *fname, MonoImageOpenStatus *status)
{
	g_return_val_if_fail (fname != NULL, NULL);

        ImageOpenOptions options = {0, }; /* zero initialize a local so all flags are false */
        options.load_options.care_about_pecoff = 1;
        return do_mono_image_open (mono_alc_get_default (), fname, status, &options);
}
```

Reply:
Thanks! Changed to pass the Options structs. Removed `mono_image_open_mibc` and opted to add `gboolean not_executable` to `mono_image_open_a_lot` and `mono_image_open_a_lot_parameterized`.